File: backend/CRUD/src/main/java/org/example/crud/dto/OfferFilter.java
package org.example.crud.dto;

import jakarta.persistence.ManyToOne;
import lombok.Data;
import org.example.crud.entity.User;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Data
public class OfferFilter {

    private String customerName;
    private String username;
    private LocalDate date;

    private List<String> productsNames;

    private BigDecimal price;
    private BigDecimal priceFrom;
    private BigDecimal priceTo;

    private BigDecimal discountPrice;
    private BigDecimal discountPriceFrom;
    private BigDecimal discountPriceTo;

    private Integer quantity;
    private Integer quantityFrom;
    private Integer quantityTo;

    private Double discountPercent;
    private Double discountPercentFrom;
    private Double discountPercentTo;


}

File: frontend/my-app/src/types/offerFilter.ts
// src/types/offerFilter.ts

export interface OfferFilter {
  customerName?: string;
  username?: string;
  date?: string;                    // "YYYY-MM-DD"
  productsNames?: string[];         // список имён продуктов для фильтрации

  price?: number;
  priceFrom?: number;
  priceTo?: number;

  discountPrice?: number;
  discountPriceFrom?: number;
  discountPriceTo?: number;

  quantity?: number;
  quantityFrom?: number;
  quantityTo?: number;

  discountPercent?: number;
  discountPercentFrom?: number;
  discountPercentTo?: number;
}
